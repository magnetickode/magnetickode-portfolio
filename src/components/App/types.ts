import { MutableRefObject } from "react";

export interface Props {
  navbarRef: MutableRefObject<HTMLDivElement | null>;
}
