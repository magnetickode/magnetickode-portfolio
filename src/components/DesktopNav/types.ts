export interface Props {
  sticky: boolean;
}

export interface StyledDesktopNavProps {
  sticky: boolean;
}

export interface StyledNavLinkProps {
  sticky: boolean;
}
