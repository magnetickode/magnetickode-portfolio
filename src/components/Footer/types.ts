export interface Props {
  footerState: string;
  footerPosition: string;
}

export interface StyledFooterProps {
  footerState: string;
  footerPosition: string;
}
