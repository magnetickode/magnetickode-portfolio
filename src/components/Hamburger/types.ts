export interface Props {
  opened: boolean;
  toggleHamburger: () => void;
}

export interface StyledHamburgerProps {
  opened: boolean;
}
