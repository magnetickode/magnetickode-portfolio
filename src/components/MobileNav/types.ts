export interface Props {
  collapse: boolean;
  hashRoute: string;
}

export interface StyledMobileNavProps {
  collapse: boolean;
}

export interface StyledNavLinkProps {
  active: boolean;
}
