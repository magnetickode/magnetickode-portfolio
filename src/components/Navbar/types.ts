import { MutableRefObject } from "react";

import { ChangeThemeGen } from "../../store/actions/types";

export interface Props {
  navbarRef?: MutableRefObject<HTMLDivElement | null>;
  isDesktop: boolean;
  stickyNavVisible: boolean;
  changeTheme: ChangeThemeGen;
}

export interface StyledNavbarProps {
  sticky: boolean;
}

export interface LogoProps {
  isDesktop: boolean;
  sticky: boolean;
}
