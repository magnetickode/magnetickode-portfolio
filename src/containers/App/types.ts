import { ScreenSizeChangedGen, ShowStickyNavGen } from "../../store/actions/types";

export interface Props {
  isDesktop: boolean;
  themeIndex: number;
  stickyNavVisible: boolean;
  screenSizeChanged: ScreenSizeChangedGen;
  showStickyNav: ShowStickyNavGen;
}
