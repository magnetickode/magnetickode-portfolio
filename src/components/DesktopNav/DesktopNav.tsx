import React from "react";
import styled from "styled-components";
import { NavHashLink as NavLink } from "react-router-hash-link";

import { Props, StyledNavLinkProps } from "./types";

const StyledDesktopNav = styled.div`
  display: flex;
  flex-basis: 70%;
  border: 0.25rem ${({ theme }) => theme.textColor} solid;
  border-radius: 0.6rem;
  user-select: none;
  overflow: hidden;
`;

const StyledNavLink = styled.div<StyledNavLinkProps>`
  flex-basis: 20%;
  height: 4rem;
  transition: all 0.3s;
  cursor: pointer;

  & a {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    font-weight: bold;
    background: ${({ active, theme }) => active && theme.textColor};
    color: ${({ active, theme }) => (active ? theme.primaryColor : theme.textColor)};
    text-decoration: none;
  }

  & a:hover {
    background: ${({ theme }) => theme.textColor};
    color: ${({ theme }) => theme.primaryColor};
  }
`;

const DesktopNav: React.FC<Props> = ({ hashRoute }) => (
  <StyledDesktopNav data-test="DesktopNavComponent">
    <StyledNavLink active={hashRoute === "#home"}>
      <NavLink smooth exact to="/#home">
        Home
      </NavLink>
    </StyledNavLink>
    <StyledNavLink active={hashRoute === "#skills"}>
      <NavLink smooth to="/#skills">
        Skills
      </NavLink>
    </StyledNavLink>
    <StyledNavLink active={hashRoute === "#portfolio"}>
      <NavLink smooth to="/#portfolio">
        Portfolio
      </NavLink>
    </StyledNavLink>
    <StyledNavLink active={hashRoute === "#about"}>
      <NavLink smooth to="/#about">
        About
      </NavLink>
    </StyledNavLink>
    <StyledNavLink active={hashRoute === "#contact"}>
      <NavLink smooth to="/#contact">
        Contact
      </NavLink>
    </StyledNavLink>
  </StyledDesktopNav>
);

export default DesktopNav;
