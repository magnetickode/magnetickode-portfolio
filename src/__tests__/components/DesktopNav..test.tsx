import React from "react";
import { shallow, ShallowWrapper } from "enzyme";

import DesktopNav from "../../components/DesktopNav";
import { Props } from "../../components/DesktopNav/types";

let wrapper: ShallowWrapper;

const props: Props = {
  sticky: false
};

beforeEach(() => {
  wrapper = shallow(<DesktopNav {...props} />);
});

it("matches snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});

it("renders without error", () => {
  const DesktopNavComponent = wrapper.find("[data-test='DesktopNavComponent']");
  expect(DesktopNavComponent.length).toBe(1);
});
