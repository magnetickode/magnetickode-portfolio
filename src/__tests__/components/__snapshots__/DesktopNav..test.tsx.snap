// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<styled.div
  data-test="DesktopNavComponent"
  sticky={false}
>
  <styled.div
    sticky={false}
  >
    <NavLink
      exact={true}
      to="/"
    >
      Home
    </NavLink>
  </styled.div>
  <styled.div
    sticky={false}
  >
    <NavLink
      to="/skills"
    >
      Skills
    </NavLink>
  </styled.div>
  <styled.div
    sticky={false}
  >
    <NavLink
      to="/portfolio"
    >
      Portfolio
    </NavLink>
  </styled.div>
  <styled.div
    sticky={false}
  >
    <NavLink
      to="/about"
    >
      About
    </NavLink>
  </styled.div>
  <styled.div
    sticky={false}
  >
    <NavLink
      to="/contact"
    >
      Contact
    </NavLink>
  </styled.div>
</styled.div>
`;
